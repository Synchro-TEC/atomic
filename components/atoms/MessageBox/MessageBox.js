import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const MessageBox = ({ theme, className, title, onClose, children, ...otherProps }) => {
  const baseClass = `sv-messagebox ${theme}`;
  const rootClass = cx(baseClass, className);

  return (
    <div className={rootClass} {...otherProps}>
      <button className="sv-messagebox__close" onClick={onClose}>
        &times;
      </button>
      <header>
        <h6>{title}</h6>
      </header>
      <main>{children}</main>
    </div>
  );
};

MessageBox.defaultProps = {
  theme: 'info',
};

MessageBox.propTypes = {
  onClose: PropTypes.func,
  title: PropTypes.string,
  theme: PropTypes.oneOf(['info', 'success', 'error', 'warning']),
};

export default MessageBox;
