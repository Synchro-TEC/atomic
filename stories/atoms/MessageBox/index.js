import React from 'react';
import { text, boolean, object } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { optionalSelect } from '../../../utils/optionalSelect';

import readme from './README.md';
import MessageBox from '../../../components/MessageBox';

const options = {
  info: 'info',
  success: 'success',
  warning: 'warning',
  error: 'error',
};

const component = () => (
  <MessageBox
    theme={optionalSelect('Theme', options, '')}
    onClose={action('close clicked')}
    title={text('Title', 'Title')}
  >
    Message
  </MessageBox>
);

export default [readme, component];
