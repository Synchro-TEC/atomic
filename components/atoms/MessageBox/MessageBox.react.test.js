import React from 'react';
import MessageBox from './MessageBox';

describe('MessageBox Component', () => {
  test('Render MessageBox', () => {
    const wrapper = shallow(<MessageBox>My MessageBox</MessageBox>);
    expect(wrapper).toMatchSnapshot();
  });

  test('Fire onClose Event', () => {
    const mockCallBack = jest.fn();
    const closeButton = shallow(<MessageBox onClose={mockCallBack}>My MessageBox</MessageBox>);
    closeButton.find('button.sv-messagebox__close').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
