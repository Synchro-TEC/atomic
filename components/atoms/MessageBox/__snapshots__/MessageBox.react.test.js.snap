// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox Component Render MessageBox 1`] = `
<div
  className="sv-messagebox info"
>
  <button
    className="sv-messagebox__close"
  >
    Ã—
  </button>
  <header>
    <h6 />
  </header>
  <main>
    My MessageBox
  </main>
</div>
`;
